docs/testing-styleguide.MD

```diff
@@ -0,0 +1,278 @@
+
+You should use snapshots in tests if amendments are being made to non-styled components to check styling is applied, but where existing and new styled components are concerned the use of snapshots is left up to the developer to decide if there is value. If you do want to use them, ensure they are small, focussed and effective.
+
+#### Continuous Integration (CI)
+The Travis CI service runs unit tests at every commit/push/pull request creation. You can manually run the tests with:
+1.	`npm run lint` - runs lint test.
+2.	`npm test` - runs jest tests.
+3.	Results can be investigated from the terminal or from the Travis dashboard linked from the branch/pull request checks
+
+### Functional Browser Testing
+#### Manual Testing
+Every new or amended component should be fully tested in either Storybook and/or [codesandbox.io](https://codesandbox.io) using functional testing, UI testing and internationalization testing techniques. We use [Jira](https://www.atlassian.com/software/jira) for issue and project tracking. 
+
+##### Browser Support
+Carbon is supported in the following browsers:
+* Chrome
+* Firefox
+* Safari
+* Edge Chromium
+
+Components should be tested in all supported browsers unless the piece of functionality is not specifically related to a UI feature, e.g. decimal precision.
+
+##### Jira
+For every testable task, a test session needs to be created for each of the supported browsers. Results should be added to each of the conditions in the created test sessions with a pass `(/)` or fail `(x)` mark.
+
+Any issues found during testing should be raised as a "Defect" in Jira unless the ticket is already of a “Bug” type, in which case comments and screenshots/recordings/GIFs can be added to the existing ticket.
+
+If the issue is present on the master branch you should first check if the issue has already been logged in GitHub. If it is present, it should be upvoted with a “+1” comment. If it has not been logged, details of how to do this are contained in the [CONTRIBUTING.MD](https://github.com/Sage/carbon/blob/master/CONTRIBUTING.md) guide. Collaborators internal to Sage should also log a Jira ticket for the issue as a "Bug" type with the GitHub issue linked. However, there should be no link from the GitHub issue to the Jira ticket.
+
+##### Storybook
+Storybook forms our Carbon component library documentation. The component stories are written in `MDX` format containing as many examples as required to showcase our components and with minimal knobs for changing prop states. Every component should have at least one story (named `default`/`multiple`/`validation` etc.) located in the `Design System` directory with `Docs`, `Notes` and `Accessibility` sections. Where required, it should be possible to test all events, e.g. `onClick`, `drag` etc, in these stories. Results of events tests can be found in the `Actions` tab.
```

Comment:
```suggestion
Storybook forms our Carbon component library documentation. The component stories are written in `MDX` format containing as many examples as required to showcase our components and with minimal knobs for changing prop states. Every component should have at least one story (named `default`/`multiple`/`validation` etc.) located in the `Design System` directory. Where required, it should be possible to test all events, e.g. `onClick`, `drag` etc, in these stories. Results of events tests can be found in the `Actions` tab.
```
MDX populates the `Docs` automatically. `Docs` is replacing `Notes` and `Accessibility` is automatically available in canvas view.